From e71fbd525561dd792f3b5f55ddce5da995d75250 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 16:13:30 +0100
Subject: [PATCH 1/7] FIX: Fix unordered_set declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/libslic3r/LogSink.hpp: At global scope:
/run/build/BambuStudio/src/libslic3r/LogSink.hpp:32:10: error: ‘unordered_set’ in namespace ‘std’ does not name a template type
   32 |     std::unordered_set<std::string> m_log_files;// the generated log files
      |          ^~~~~~~~~~~~~
/run/build/BambuStudio/src/libslic3r/LogSink.hpp:11:1: note: ‘std::unordered_set’ is defined in header ‘<unordered_set>’; this is probably fixable by adding ‘#include <unordered_set>’
   10 | #include "Utils.hpp"
  +++ |+#include <unordered_set>
   11 |
---
 src/libslic3r/LogSink.cpp | 2 --
 src/libslic3r/LogSink.hpp | 1 +
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/libslic3r/LogSink.cpp b/src/libslic3r/LogSink.cpp
index 05f0223b32bf..1aa84dfd5e91 100644
--- a/src/libslic3r/LogSink.cpp
+++ b/src/libslic3r/LogSink.cpp
@@ -12,8 +12,6 @@
 #include <nlohmann/json.hpp>
 #include <openssl/aes.h>
 
-#include <unordered_set>
-
 //#define DEBUG_LOG_ENC
 
 #define HEADER_BEGIN_MARKER "BEGIN_HEADER\n"
diff --git a/src/libslic3r/LogSink.hpp b/src/libslic3r/LogSink.hpp
index d1b092a9809c..1d321c848112 100644
--- a/src/libslic3r/LogSink.hpp
+++ b/src/libslic3r/LogSink.hpp
@@ -2,6 +2,7 @@
 
 #include <string>
 #include <ctime>
+#include <unordered_set>
 
 #include <boost/log/sinks/basic_sink_backend.hpp>
 #include <boost/log/sinks/sync_frontend.hpp>
-- 
2.52.0


From c6332ce4c1559ad2fdd6b4fa4b6c34cbf5315d71 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 16:16:40 +0100
Subject: [PATCH 2/7] FIX: Fix undeclared boost::filesystem
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/libslic3r/LogSink.cpp:436:33: error: ‘exists’ is not a member of ‘boost::filesystem’; did you mean ‘std::filesystem::exists’?
  436 |         if (!boost::filesystem::exists(log_folder)) {
      |                                 ^~~~~~
---
 src/libslic3r/LogSink.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libslic3r/LogSink.cpp b/src/libslic3r/LogSink.cpp
index 1aa84dfd5e91..8dc39a5f3d3a 100644
--- a/src/libslic3r/LogSink.cpp
+++ b/src/libslic3r/LogSink.cpp
@@ -5,6 +5,7 @@
 #include "slic3r/Utils/BBLUtil.hpp"
 #include "slic3r/Utils/Http.hpp"
 
+#include <boost/filesystem.hpp>
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/uuid_generators.hpp>
 #include <boost/uuid/uuid_io.hpp>
-- 
2.52.0


From 5989085c5705fc22de8d7b36f90ab9a8647264b6 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 16:19:46 +0100
Subject: [PATCH 3/7] FIX: Fix undeclared std::ofstream
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/libslic3r/LogSink.cpp:395:23: error: variable ‘std::ofstream ofs’ has initializer but incomplete type
  395 |     std::ofstream ofs(output_path, std::ios::binary | std::ios::trunc);
      |                       ^~~~~~~~~~~
/run/build/BambuStudio/src/libslic3r/LogSink.cpp:395:23: note: ‘std::ofstream’ is defined in header ‘<fstream>’; this is probably fixable by adding ‘#include <fstream>’
---
 src/libslic3r/LogSink.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libslic3r/LogSink.cpp b/src/libslic3r/LogSink.cpp
index 8dc39a5f3d3a..649f304911f6 100644
--- a/src/libslic3r/LogSink.cpp
+++ b/src/libslic3r/LogSink.cpp
@@ -13,6 +13,8 @@
 #include <nlohmann/json.hpp>
 #include <openssl/aes.h>
 
+#include <fstream>
+
 //#define DEBUG_LOG_ENC
 
 #define HEADER_BEGIN_MARKER "BEGIN_HEADER\n"
-- 
2.52.0


From d92b41977f771a9d5a06e3c90e1585e50fb85c43 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 16:29:39 +0100
Subject: [PATCH 4/7] FIX: Fix undeclared BOOST_LOG_TRIVIAL
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/libslic3r/ObjColorUtils.cpp:304:9: error: ‘BOOST_LOG_TRIVIAL’ was not declared in this scope
  304 |         BOOST_LOG_TRIVIAL(warning) << __FUNCTION__ << ": all colors are undefined";
      |         ^~~~~~~~~~~~~~~~~
---
 src/libslic3r/ObjColorUtils.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libslic3r/ObjColorUtils.cpp b/src/libslic3r/ObjColorUtils.cpp
index 0cbdd13c6fc2..7963cffb375e 100644
--- a/src/libslic3r/ObjColorUtils.cpp
+++ b/src/libslic3r/ObjColorUtils.cpp
@@ -1,3 +1,5 @@
+#include <boost/log/trivial.hpp>
+
 #include "ObjColorUtils.hpp"
 #include "Model.hpp"
 #include "admesh/stl.h"
-- 
2.52.0


From 10101f73e69fbae1262aa50b616282057055e79a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 16:43:51 +0100
Subject: [PATCH 5/7] FIX: Fix undeclared item_12h_time_format

item_12h_time_format is not instantiated anywhere, remove it.
---
 src/slic3r/GUI/Preferences.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/slic3r/GUI/Preferences.cpp b/src/slic3r/GUI/Preferences.cpp
index 4243a1c57147..8c5121debbc5 100644
--- a/src/slic3r/GUI/Preferences.cpp
+++ b/src/slic3r/GUI/Preferences.cpp
@@ -1458,7 +1458,6 @@ wxWindow* PreferencesDialog::create_general_page()
     sizer_page->Add(item_language, 0, wxTOP, FromDIP(3));
     sizer_page->Add(item_region, 0, wxTOP, FromDIP(3));
     sizer_page->Add(item_currency, 0, wxTOP, FromDIP(3));
-    sizer_page->Add(item_12h_time_format, 0, wxTOP, FromDIP(3));
     sizer_page->Add(item_auto_flush, 0, wxTOP, FromDIP(3));
     sizer_page->Add(item_single_instance, 0, wxTOP, FromDIP(3));
     sizer_page->Add(item_bed_type_follow_preset, 0, wxTOP, FromDIP(3));
-- 
2.52.0


From f0f592a40df91ea50730449b4e29fd8244a7878d Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 17:01:45 +0100
Subject: [PATCH 6/7] FIX: Fix undeclared std::mt19937
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/slic3r/Utils/BBLUtil.cpp: In static member function ‘static std::string Slic3r::BBLCrossTalk::Encode_DevIp(const std::string&, const std::string&)’:
/run/build/BambuStudio/src/slic3r/Utils/BBLUtil.cpp:228:10: error: ‘mt19937’ is not a member of ‘std’
  228 |     std::mt19937 rng(seed);
      |          ^~~~~~~
---
 src/slic3r/Utils/BBLUtil.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/slic3r/Utils/BBLUtil.cpp b/src/slic3r/Utils/BBLUtil.cpp
index 3b9fc8f0d047..700318333d79 100644
--- a/src/slic3r/Utils/BBLUtil.cpp
+++ b/src/slic3r/Utils/BBLUtil.cpp
@@ -1,6 +1,8 @@
 #include "BBLUtil.hpp"
 #include "libslic3r_version.h"
 
+#include <random>
+
 #include <boost/log/trivial.hpp>
 
 #include <openssl/aes.h>
-- 
2.52.0


From 9fe63e43b9fc491b75a6e312ceaea16f7542b29b Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Jan 2026 17:07:58 +0100
Subject: [PATCH 7/7] FIX: Fix undeclared std::optional
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In file included from /run/build/BambuStudio/src/slic3r/GUI/DeviceCore/DevFilaBlackList.cpp:3:
/run/build/BambuStudio/src/slic3r/GUI/DeviceCore/DevFilaBlackList.h:24:14: error: ‘optional’ in namespace ‘std’ does not name a template type
   24 |         std::optional<bool> used_for_print_support;// optional
      |              ^~~~~~~~
/run/build/BambuStudio/src/slic3r/GUI/DeviceCore/DevFilaBlackList.h:5:1: note: ‘std::optional’ is defined in header ‘<optional>’; this is probably fixable by adding ‘#include <optional>’
    4 | #include "slic3r/Utils/json_diff.hpp"
  +++ |+#include <optional>
    5 |
---
 src/slic3r/GUI/DeviceCore/DevFilaBlackList.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/slic3r/GUI/DeviceCore/DevFilaBlackList.h b/src/slic3r/GUI/DeviceCore/DevFilaBlackList.h
index 65b975fcc5df..163a8d759403 100644
--- a/src/slic3r/GUI/DeviceCore/DevFilaBlackList.h
+++ b/src/slic3r/GUI/DeviceCore/DevFilaBlackList.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <optional>
 #include <wx/string.h>
 #include "slic3r/Utils/json_diff.hpp"
 
-- 
2.52.0

